### PR TITLE
Remaining tests

### PR DESCRIPTION
**Pull request by pablisco**

### CODE REVIEW

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt [hunk 1]
```diff
@@ -121,4 +124,61 @@ class RoomEventsSpek : Spek({
...
+            setUpInstanceWith(newUsers(PUSHERINO, ALICE))
+
+            val pusherino = chatFor(PUSHERINO).connect().wait().assumeSuccess()
+
+            val room = pusherino.createRoom(GENERAL).wait().assumeSuccess()
+
+            assertThat(room.name).isEqualTo(GENERAL)
+        }
+
+        xit("updates room name", reason = "Requires elevated access") {
```
Comment (hamchapman) [inline chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: What's this one blocked on (assuming that's what the `xit` means)?

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt [hunk 2]
```diff
@@ -121,4 +124,61 @@ class RoomEventsSpek : Spek({
...
+            setUpInstanceWith(newUsers(PUSHERINO, ALICE), newRoom(GENERAL, PUSHERINO, ALICE))
+
+            val pusherino = chatFor(SUPER_USER).connect().wait().assumeSuccess()
+
+            val room = pusherino.updateRoom(pusherino.generalRoom, NOT_GENERAL).wait()
+
+            check(room is Success) { (room as? Failure)?.error as Any }
+        }
+
+        xit("deletes room", reason = "Requires elevated access") {
```
Comment (hamchapman) [inline chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: Same, what's this one blocked on?

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatkitTokenProvider.kt [hunk 3]
```diff
@@ -41,56 +42,45 @@ class ChatkitTokenProvider
...
+    private val httpUrl: Result<HttpUrl, Error>
+        get() = HttpUrl.parse(endpoint)?.newBuilder()?.apply {
+            addQueryParameter("user_id", userId)
+        }?.build().orElse { Errors.network("Incorrect endpoint: $endpoint") }
 
-        val request = Request.Builder()
-            .url(urlBuilder.build())
-            .post(requestBody)
-            .build()
+    //TODO: Maybe add auth data and params to query params instead of body?
```
Comment (hamchapman) [inline chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatkitTokenProvider.kt]: OAuth2 spec sends it as part of the body, so think we'll be leaving it as that for now

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt (hunk 1 above)
Comment (pablisco) [reply chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: This requires admin access. I didn't manage to get it to have admin rights :/

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt (hunk 2 above)
Comment (pablisco) [reply chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: Same as the one before

File: chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt (hunk 1 above)
Comment (hamchapman) [reply chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt]: There's an `admin` role automatically created for every chatkit instance, so you could assign that to one of your users and use that to delete / update rooms

**Pull request merged**